src/login/logind-session-device.c

```diff
@@ -478,3 +479,24 @@ unsigned int session_device_try_pause_all(Session *s) {
 
         return num_pending;
 }
+
+int session_restore_devices(Session *s, const char *devices) {
+        const char *word, *state;
+        size_t l;
+        int r;
+
+        assert(s);
+
+        if (devices)
+                FOREACH_WORD(word, l, devices, state) {
+                        SessionDevice *sd;
+                        dev_t dev = strtoul(word, NULL, 10);
```

Comment:
please don't use "strtoul()", use something like safe_atou32() or so (or even better, add a parse_dev() or so, which does this check properly.

Also, don't mix variable declarations and function invocations, see CODING_STYLE.